aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -158,29 +168,117 @@ public void configureAndInitializeInApiModeWithoutApi() throws JSONException, Am
         }
     }
 
-    private void assertSyncProcessorStarted() {
-        boolean syncProcessorInvoked = mockingDetails(modelProvider)
-            .getInvocations()
-            .stream()
-            .anyMatch(invocation -> invocation.getLocation().getSourceFile().contains("SyncProcessor"));
+    /**
+     * Verify that when the clear method is called, the following happens
+     * - All remote synchronization processes are stopped
+     * - The database is deleted.
+     * - On the next interaction with the DataStore, the synchronization processes are restarted.
+     * @throws JSONException on failure to arrange plugin config
+     * @throws AmplifyException on failure to arrange API plugin via Amplify facade
+     */
+    @Test
+    public void clearStopsSyncUntilNextInteraction() throws AmplifyException, JSONException {
+        ApiCategory mockApiCategory = mockApiCategoryWithGraphQlApi();
+        JSONObject dataStorePluginJson = new JSONObject()
+            .put("syncIntervalInMinutes", 60);
+        AWSDataStorePlugin awsDataStorePlugin = new AWSDataStorePlugin(modelProvider, mockApiCategory);
+        awsDataStorePlugin.configure(dataStorePluginJson, context);
+        awsDataStorePlugin.initialize(context);
+
+        // Trick the DataStore since it's not getting initialized as part of the Amplify.initialize call chain
+        Amplify.Hub.publish(HubChannel.DATASTORE, HubEvent.create(InitializationStatus.SUCCEEDED));
+
+        assertRemoteSubscriptionsStarted();
+
+        Person person1 = createPerson("Test", "Dummy I");
+        Person person2 = createPerson("Test", "Dummy II");
+        Completable.fromSingle(single -> { // Save a record to local store
+            awsDataStorePlugin.save(person1, itemSaved -> {
+                assertNotNull(itemSaved.item().getId());
+                assertEquals(person1.getLastName(), itemSaved.item().getLastName());
+                single.onSuccess(true);
```

Comment:
Since you ignore the result of the single, can't you just use `Completable.create`, `Completable.fromAction`, `Completable.fromRunnable`, etc?